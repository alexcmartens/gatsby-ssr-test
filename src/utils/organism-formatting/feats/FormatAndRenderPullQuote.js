import React, { useState } from "react"
import { formatFocusedImage } from "../formatBaseMolecules"
import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import formatPageLinks from "../formatPageLinks"
import formatVideo from "../formatVideo"
import { PullQuote } from "terra-one"
import { Link } from "gatsby"
import formatLottieAnimation from "../formatLottieAnimation"

const FormatAndRenderPullQuote = props => {
  const [override, setOverride] = useState(false)
  const { data = {}, i, locale, products, siteIdentifier } = props

  const text = data?.quote
  const source = data?.author
  const sourceTitle = data?.authorCredentials
  const quote = { text, source, sourceTitle }
  const imagePosition = data?.imagePosition?.toString() || "Left"
  const appearance = data?.appearance || "Light"
  const imageOptions = data?.imageOptions || "1/2"
  const pullQuoteLottieStyles = {}
  const topImage = formatPullQuoteImage(data?.topImage, pullQuoteLottieStyles)
  const sideImage = formatPullQuoteImage(data?.sideImage, pullQuoteLottieStyles)
  const backgroundImage = formatPullQuoteImage(
    data?.backgroundImage,
    pullQuoteLottieStyles
  )

  const linkType = data?.cta?.url?.contentType
    ? data?.cta?.url?.contentType
    : data?.cta?.url?.internal?.type
    ? data?.cta?.url?.internal?.type
    : null

  const cta = formatPageLinks(
    data?.cta,
    locale,
    linkType,
    null,
    props?.currentSpaceId,
    products,
    siteIdentifier
  )

  if (cta?.video) {
    cta.video = formatVideo(cta.video)
    cta["override"] = override
    cta["setOverride"] = setOverride
  }
  const theme =
    appearance === "Dark"
      ? {
          backgroundColor: "#252a2e",
          color: "#fff",
          ctaClass: "secondary--1",
        }
      : {
          backgroundColor: "#fff",
          color: "#252a2e",
          ctaClass: "primary--2",
        }

  if (cta) {
    cta.className = theme?.ctaClass
  }

  const content = {
    quote,
    cta,
    theme,
    Link,
    topImage,
    sideImage,
    imageOptions,
    imagePosition,
    appearance,
    backgroundImage,
  }

  const isTopImage = topImage?.focalPointImage?.url

  const variant = () => {
    if (isTopImage) {
      return "variant3"
    } else if (imageOptions === "1/2") {
      return "variant2"
    } else if (imageOptions === "1/4") {
      if (appearance === "Light") {
        return "variant1"
      }
      if (appearance === "Dark") {
        return "variant2"
      }
    } else if (imageOptions === "No Image") {
      if (appearance === "Light") {
        return "variant1"
      }
      if (appearance === "Dark") {
        return "variant2"
      }
    } else {
      return ""
    }
  }

  const padding = true,
    fullBleedColor = false,
    inheritBackground = false

  return renderPaddingOrFullBleed(
    <PullQuote content={content} variant={variant()} key={i} />,
    padding,
    fullBleedColor,
    inheritBackground
  )
}

export default FormatAndRenderPullQuote

const formatPullQuoteImage = (image, styles) => {
  return image?.__typename === "ContentfulLottieAnimation"
    ? formatLottieAnimation(image?.json || image?.srcLink, styles)
    : formatFocusedImage(image)
}
