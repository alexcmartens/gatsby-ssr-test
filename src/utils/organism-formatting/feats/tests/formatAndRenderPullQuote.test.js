import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderPullQuote from "../FormatAndRenderPullQuote"

jest.mock("../../formatAnchor")
jest.mock("../../formatTheme")
jest.mock("../../CTASectionFormat")
jest.mock("../../formatBaseMolecules")
jest.mock("../../ctaFormat")

describe("FormatAndRenderPullQuote", () => {
  it("renders correctly", () => {
    const content = {
      __typename: "ContentfulPullQuote",
      title: "Pull Quote Test",
      quote:
        "Pull Quote can be aliquip aliqua aliqua id qui aute anim reprehenderit tempor velit est sint laborum. Cillum et laboris aliquip consequat",
      author: "Gariatur Nostrud Pariatur",
      authorCredentials: "TITLE OR CREDENTIALS FOR THE QUOTEE",
      imageWidth: "1/2",
      imagePosition: "Right",
      appearance: "Dark",
      cta: { text: "Call To Action", url: "https://example.com" },
      topImage: null,
      sideImage: {
        title: "720x720",
        image: {
          title: "720x720",
          file: {
            url: "//images.ctfassets.net/citn2sn5tdjr/2aIM2EvXTK5yrpIHrvP5k5/6f7a3669abb2bf2ec39b635a91da90f6/720x720.jpg",
            details: { image: { height: 720, width: 720 } },
          },
        },
        focalPoint: { focalPoint: { y: 224, x: 358 } },
      },
    }

    const tree = renderer
      .create(<FormatAndRenderPullQuote data={content} i={0} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
