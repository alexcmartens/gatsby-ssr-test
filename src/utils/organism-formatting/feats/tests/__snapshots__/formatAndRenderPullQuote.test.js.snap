// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderPullQuote renders correctly 1`] = `
<section
  className="default-layout   mockconstructor
      default-layout--no-bottom-margin"
  style={
    Object {
      "backgroundColor": "",
      "marginBottom": "",
    }
  }
>
  <section
    className="default-layout__grid 
          
        "
    style={
      Object {
        "backgroundColor": "",
        "marginBottom": "",
      }
    }
  >
    <div
      className="default-layout__children "
    >
      <div
        content={
          Object {
            "Link": [MockFunction],
            "appearance": "Dark",
            "backgroundImage": Object {
              "alt": "River",
              "focalPointImage": Object {
                "height": 500,
                "title": "River",
                "url": "//images.ctfassets.net/citn2sn5tdjr/7zF5J9FlzLzm2SFrT6inNJ/2fe28905a10bd27e8d6b178d3a8239e6/save.png",
                "width": 500,
                "x": 490,
                "y": 323,
              },
            },
            "cta": Object {
              "Link": [MockFunction],
              "className": "secondary--1",
              "iconImage": undefined,
              "override": null,
              "setOverride": null,
              "text": "Call To Action",
              "url": "/en/undefined",
              "video": null,
            },
            "imagePosition": "Right",
            "imageWidth": "1/2",
            "quote": Object {
              "source": "Gariatur Nostrud Pariatur",
              "sourceTitle": "TITLE OR CREDENTIALS FOR THE QUOTEE",
              "text": "Pull Quote can be aliquip aliqua aliqua id qui aute anim reprehenderit tempor velit est sint laborum. Cillum et laboris aliquip consequat",
            },
            "sideImage": Object {
              "alt": "River",
              "focalPointImage": Object {
                "height": 500,
                "title": "River",
                "url": "//images.ctfassets.net/citn2sn5tdjr/7zF5J9FlzLzm2SFrT6inNJ/2fe28905a10bd27e8d6b178d3a8239e6/save.png",
                "width": 500,
                "x": 490,
                "y": 323,
              },
            },
            "theme": Object {
              "backgroundColor": "#252a2e",
              "color": "#fff",
              "ctaClass": "secondary--1",
            },
            "topImage": Object {
              "alt": "River",
              "focalPointImage": Object {
                "height": 500,
                "title": "River",
                "url": "//images.ctfassets.net/citn2sn5tdjr/7zF5J9FlzLzm2SFrT6inNJ/2fe28905a10bd27e8d6b178d3a8239e6/save.png",
                "width": 500,
                "x": 490,
                "y": 323,
              },
            },
          }
        }
      />
    </div>
  </section>
</section>
`;
